partiql-parser/src/main/java/org/partiql/parser/ParserConfig.java

```diff
@@ -0,0 +1,18 @@
+package org.partiql.parser;
+
+import org.jetbrains.annotations.NotNull;
+import org.partiql.spi.errors.PErrorListener;
+
+/**
+ * Represents the configuration for the {@link PartiQLParser}.
+ *
+ * @see PartiQLParser
+ * @see ParserConfigBuilder
+ */
+public interface ParserConfig {
+    /**
+     * @return The error listener to be used by the compiler.
+     */
+    @NotNull
+    PErrorListener getErrorListener();
+}
```

Comment:
What are some reasons not to add listeners via constructors (or even mutable class methods) like ANTLR has. These configs add a several more classes to juggle and thread through all calls, which seems like additional complexity but I may be missing how these are more useful than just adding listeners via the builders.

Reply:
See https://github.com/partiql/partiql-lang-kotlin/pull/1615#discussion_r1805257316.